src/components/Chat/AnonymousInputMessage.tsx

```diff
@@ -66,23 +73,137 @@ const styles = StyleSheet.create({
+  };
+
+  const handleUploadCamera = async (path) => {
+    setIsLoadingUploadImageCamera(true);
+
+    const resultUrls = [];
+    const uploadedImageUrl = await ImageUtils.uploadImageWithoutAuth(path);
+    resultUrls.push({
+      type: 'image',
+      asset_url: uploadedImageUrl,
+      thumb_url: uploadedImageUrl,
+      myCustomField: 'image'
+    });
+
+    onSendButtonClicked(' ', resultUrls);
+    setIsLoadingUploadImageCamera(false);
+  };
+
+  const openSettingApp = () => {
+    Linking.openSettings();
+  };
+
+  const openAlertPermission = (message) => {
+    Alert.alert('Permission Denied', message, [
+      {text: 'Open Settings', onPress: openSettingApp},
+      {text: 'Close'}
+    ]);
+  };
+
+  const onOpenMedia = async () => {
+    const {success} = await requestExternalStoragePermission();
+    if (success) {
+      ImagePicker.openPicker({
+        width: 512,
+        height: 512,
+        cropping: true,
+        multiple: true,
+        maxFiles: 20
+      }).then(async (images) => {
+        const allPromises: Promise<void>[] = images.map(async (image) => {
+          const imageCropped = await ImagePicker.openCropper({
+            path: image.path,
+            width: 512,
+            height: 512
+          });
+          return imageCropped.path;
+        });
+
+        const newImages = await Promise.all(allPromises);
+
+        handleUploadMedia(newImages, 'gallery');
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your photos, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
+
+  const onOpenCamera = async () => {
+    const {success} = await requestCameraPermission();
+    if (success) {
+      ImagePicker.openCamera({
+        width: 512,
+        height: 512,
+        cropping: true,
+        mediaType: 'photo'
+      }).then((imageRes) => {
+        handleUploadCamera(imageRes.path);
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your camera, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
+
+  const onOpenFile = async () => {
+    const {success} = await requestExternalStoragePermission();
+    if (success) {
+      DocumentPicker.pickSingle({
+        presentationStyle: 'fullScreen',
+        copyTo: 'cachesDirectory'
+      }).then((pickerResult) => {
+        console.warn('pickerResult', pickerResult);
+        //
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your document library, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
 
   const onChangeInput = (v) => {
     setText(v);
   };
+
   const onSelectEmoji = () => {
-    refEmoji.current.close();
+    refEmoji.current.open();
+  };
+
+  const onSelectAttachment = () => {
+    refAttachment.current.open();
   };
 
   const handleSendMessage = () => {
-    onSendButtonClicked(text);
+    onSendButtonClicked(text, []);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `handleSendMessage` now sends an empty array as the second argument to `onSendButtonClicked`. Ensure that this change doesn't break existing functionality where `onSendButtonClicked` was expecting only one argument.
```diff
- onSendButtonClicked(text);
+ onSendButtonClicked(text, []);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->